src/material-experimental/mdc-list/list-option.html

```diff
@@ -1,25 +1,47 @@
-<!-- Save icons and unclassified content to be placed later. -->
+<!--
+  Save icons and the pseudo checkbox so that they can be re-used in the
+  template without duplication.
+-->
 <ng-template #icons>
-  <ng-content select="[mat-list-avatar],[matListAvatar],[mat-list-icon],[matListIcon]"></ng-content>
+  <ng-content select="[mat-list-avatar],[matListAvatar],[mat-list-icon],[matListIcon]">
+  </ng-content>
 </ng-template>
-<ng-template #unsortedContent>
-  <ng-content></ng-content>
+
+<ng-template #checkbox>
+  <div class="mdc-checkbox" [class.mdc-checkbox--disabled]="disabled">
```

Comment:
This is something I'm still a bit unsure about. The initial attempt of selection list seemed to use our pseudo checkbox, but it seems like we _could_ just use the markup from the MDC checkbox to rely on MDC as much as possible?

My only worry is that that we'd need to wrap the MDC checkbox styles in the list-option styles. Any thoughts on this?

Reply:
I don't see what's stopping us from using the pseudo checkbox here. It looks identical to the MDC one, aside from the animations.

Reply:
Not totally identical:
![image](https://user-images.githubusercontent.com/838736/90071804-15b62b80-dcab-11ea-8da9-9ced5ae447b7.png)

mdc checkbox on the left, pseudo checkbox on the right. The checkmark is shorted and thicker. We should see if we can actually make them identical, though, and potentially try to upstream something to MDC instead of trying to keep up with them

Reply:
> We might as well go with this more now and separately push the MDC team for some equivalent to pseudo-checkbox.

That's something they already partially already have (which is where I was going). Their components always consist of initial static HTML markup and a later "upgraded" variant (instantiated through the component/foundation).

We only use the markup here, as we do not rely on any of larger runtime code here. This seems almost equivalent to an pseudo checkbox, or sufficient to what we need here. The only difference is that the checkbox is still driven by an native input. This should not be an issue though as we do not include it in the tab order.

> problematic for accessibility since a selection list should behave as a listbox and not like a list of checkboxes.

It still can do that; as said above, we purely use it decoratively through the static markup. Also as a side-fact: MDC also
has examples of selection list w/ checkboxes but they use `role=group` and `role=checkbox` for items. I've created an issue upstream for that as that doesn't work well with screen readers.

> My concern is that the full checkbox is too heavy 

Yeah, that is the primary concern I had too. This is something though that ultimately _should_ not be a problem if we have fine-grained Sass for these components. That way we could just pull the base styles. I assume it will be still larger than our existing pseudo checkbox, but the benefit is that we share as much code as possible with MDC.

@crisbeto For sure we could use the existing pseudo checkbox. Though the point seems to rely as much as possible on MDC. We could always rely on existing non-MDC code. I don't think anything except for the current potential size overhead (that we might be able to boil down with fine-grained Sass features as per my proposal) is stopping us from using the actual MDC checkbox.

Reply:
The number of DOM and elements and SVG content is a concern of mine, but I'm fine with leaving this as-is for now and putting it on our backlog with MDC.

Reply:
It doesn't seem to expensive at first glance, compared to the other case where another new component is matched and instantiated all the time (i.e. the pseudo checkbox component). I don't feel very strong about it, and also I'm not 100% convinced of either approaches (hence this discussion). Agreed that we could look into this later as part of the backlog.